packages/app/src/debug/DebugPageDetails.vue

```diff
@@ -0,0 +1,67 @@
+<template>
+  <div
+    class="flex flex-col w-full px-24px justify-center items-center align-middle"
+    :class="{'flex-grow': ['PASSED', 'OVERLIMIT'].includes(status)}"
+  >
+    <DebugCancelledAlert
+      v-if="status === 'CANCELLED'"
+      :total-specs="totalSpecs"
+      :total-skipped-specs="totalSkippedSpecs"
+      :canceled-at="canceledAt"
+      :canceled-by-full-name="canceledByFullName"
+      :canceled-by-email="canceledByEmail"
+    />
+    <DebugPassed v-else-if="status === 'PASSED'" />
+    <DebugErrored
+      v-else-if="status === 'ERRORED'"
+      :errors="errors"
+      :total-specs="totalSpecs"
+      :total-skipped-specs="totalSkippedSpecs"
+    />
+    <DebugNoTests v-else-if="status === 'NOTESTS'" />
+    <DebugTimedout
+      v-else-if="status === 'TIMEDOUT'"
+      :total-specs="totalSpecs"
+      :total-skipped-specs="totalSkippedSpecs"
+      :ci="ci"
+    />
+    <DebugOverLimit
+      v-else-if="status === 'OVERLIMIT' || isHiddenByUsageLimits"
+      :over-limit-action-type="overLimitActionType"
+      :over-limit-action-url="overLimitActionUrl"
+    />
+  </div>
+</template>
+<script lang="ts" setup>
+import { computed } from '@vue/reactivity'
+import type { CloudRunStatus, OverLimitActionTypeEnum } from '../generated/graphql'
+import DebugCancelledAlert from './DebugCancelledAlert.vue'
+import DebugPassed from './DebugPassed.vue'
+import DebugErrored from './DebugErrored.vue'
+import DebugNoTests from './DebugNoTests.vue'
+import DebugTimedout from './DebugTimedout.vue'
+import DebugOverLimit from './DebugOverLimit.vue'
+import type { CloudCiBuildInfo } from '@packages/data-context/src/gen/graphcache-config.gen'
```

Comment:
I wonder if `packages/graphql/src/gen/cloud-source-types.gen.ts` is a more correct place to import this from (note it's from `graphql`, not `data-context` - I feel like the frontend code should **not** import anything from `data-context`, even just a type, since that's really a server package).

Reply:
I can't import from `@packages/graphql` because it's restricted by ESLint. I ended up declaring a fragment for it and using that type instead.